packages/client-common/__tests__/integration/role.test.ts

```diff
@@ -0,0 +1,359 @@
+          password,
+          role: roleName1,
+        })
+
+        await tryInsert()
+        await assertJsonValues(defaultClient, tableName)
+      },
+    )
+
+    whenOnEnv(TestEnv.LocalSingleNode).it(
+      'should successfully insert when client specifies multiple roles and at least one is allowed to insert',
+      async () => {
+        client = createTestClient({
+          username,
+          password,
+          role: [roleName1, roleName2],
+        })
+
+        await tryInsert()
+        await assertJsonValues(defaultClient, tableName)
+      },
+    )
+
+    whenOnEnv(TestEnv.LocalSingleNode).it(
+      'should fail to insert when client specifies a role that is not allowed to insert',
+      async () => {
+        client = createTestClient({
+          username,
+          password,
+          role: roleName2,
+        })
+
+        await expectAsync(tryInsert()).toBeRejectedWith(
+          jasmine.objectContaining({
+            message: jasmine.stringContaining('Not enough privileges'),
+            code: '497',
+            type: 'ACCESS_DENIED',
+          }),
+        )
+      },
+    )
+
+    whenOnEnv(TestEnv.LocalSingleNode).it(
+      'should successfully insert when insert specifies a role that is allowed to insert',
+      async () => {
+        client = createTestClient({
+          username,
+          password,
+          role: roleName2,
+        })
+
+        await tryInsert(roleName1)
+        await assertJsonValues(defaultClient, tableName)
+      },
+    )
+
+    whenOnEnv(TestEnv.LocalSingleNode).it(
+      'should successfully insert when insert specifies multiple roles and at least one is allowed to insert',
+      async () => {
+        client = createTestClient({
+          username,
+          password,
+          role: roleName2,
+        })
+
+        await tryInsert([roleName1, roleName2])
+        await assertJsonValues(defaultClient, tableName)
+      },
+    )
+
+    whenOnEnv(TestEnv.LocalSingleNode).it(
+      'should fail to insert when insert specifies a role that is not allowed to insert',
+      async () => {
+        client = createTestClient({
+          username,
+          password,
+          role: roleName1,
+        })
+
+        await expectAsync(tryInsert(roleName2)).toBeRejectedWith(
+          jasmine.objectContaining({
+            message: jasmine.stringContaining('Not enough privileges'),
+            code: '497',
+            type: 'ACCESS_DENIED',
+          }),
+        )
+      },
+    )
+  })
+
+  describe('for commands', () => {
+    let tableName: string
+
+    beforeEach(async () => {
+      tableName = `command_role_test_${guid()}`
+    })
+
+    async function tryCreateTable(role?: string | Array<string>) {
+      const query = `
+        CREATE TABLE ${tableName}
+        (id UInt64, name String, sku Array(UInt8), timestamp DateTime)
+        ENGINE = MergeTree()
+        ORDER BY (id)
+      `
+      await client.command({ query, role })
+    }
```

Comment:
Nit: you could just use `createSimpleTable` like [here](https://github.com/ClickHouse/clickhouse-js/blob/bd262b9aea9ea509b945924d882e3715f4086cb6/packages/client-common/__tests__/integration/auth.test.ts#L51-L66)

Reply:
The problem I have with that approach is that it doesn't allow me to pass through a `role` setting to add to the `.command` call in `createTable()`. I could update `createSimpleTable` and `createTable` to accept optional `command` options if you'd like, or keep it as is, or if you had a different workaround in mind please let me know

Reply:
Ah yes, I see. It's fine to leave it like this then.